examples/rnn_example.lua

```diff
@@ -10,48 +10,66 @@ local n_hidden = 25
 local batch_size = 15
 local seq_length = 5
 
-local function make_rnn(n_input, n_hidden, n_output)
-	local input = nn.Identity()()
-	local prev_state = nn.Identity()()
+local function make_rnn_layer(n_input, n_hidden)
+    local input = nn.Identity()()
+    local prev_state = nn.Identity()()
 
-	local next_state = nn.Sigmoid()(nn.CAddTable()({
-		nn.Linear(n_input, n_hidden)(input),
-		nn.Linear(n_hidden, n_hidden)(prev_state)
-	}))
+    local next_state = nn.Sigmoid()(nn.CAddTable()({
+        nn.Linear(n_input, n_hidden)(input),
+        nn.Linear(n_hidden, n_hidden)(prev_state)
+    }))
 
-	local output = nn.Linear(n_hidden, n_output)(next_state)
+    output = nn.Identity()(next_state)
 
-	return nn.gModule({input, prev_state}, {output, next_state})
+    return nn.gModule({input, prev_state}, {output, next_state})
 end
 
 local model = {
-	rnn = make_rnn(n_input, n_hidden, n_output),
-	criterion = nn.MSECriterion(),
-	start_state = torch.zeros(batch_size, n_hidden),
-
-	tape = autobw.Tape(),
-
-	forward = function(self, inputs, targets)
-		local loss = 0
-		local next_state = self.start_state
-
-		self.tape:start()
-
-		for i = 1, inputs:size(1) do
-			output, next_state = unpack(self.rnn:forward({inputs[i], next_state}))
-			loss = loss + self.criterion:forward(output, targets[i])
-		end
-
-		self.tape:stop()
-
-		self.start_state:copy(next_state)
-
-		return loss
-	end,
-
-	backward = function(self)
-		self.tape:backward()
-	end,
+    -- Add extra layers here (it doesn't help on this problem)
+    layers = {
+        make_rnn_layer(n_input, n_hidden),
+        --make_rnn_layer(n_hidden, n_hidden),
+    },
+    state = {
+        torch.zeros(batch_size, n_hidden),
+        --torch.zeros(batch_size, n_hidden),
+    },
+    output = nn.Linear(n_hidden, n_output),
+    criterion = nn.MSECriterion(),
+
+    tape = autobw.Tape(),
+
+    forward = function(self, inputs, targets)
+        local loss = 0
+
+        self.tape:start()
+
+        for t = 1, inputs:size(1) do
+            local output = inputs[t]
+            for l = 1, #self.layers do
+                output, self.state[l] = unpack(self.layers[l]:forward({output, self.state[l]}))
+            end
+            output = self.output:forward(output)
+            loss = loss + self.criterion:forward(output, targets[t])
+        end
+
+        self.tape:stop()
+
+        return loss
+    end,
+
+    backward = function(self)
+        self.tape:backward()
+    end,
+
+    get_parameters = function(self)
+        local pack = nn.Sequential()
```

Comment:
Not sure if I'm being picky, but do you think it's a bit misleading to use an nn.Sequential when things aren't sequential? I think an instance of nn.Container would be more clearly a dummy object, especially for a tutorial-style example.